### DISCUSSION TITLE
Gardens Metrics 📊

### DISCUSSION THREAD

**Discussion by kanaagawa** [1Hive/gardens#518] [Ideas]
# Summary
**Metrics that gives a visualization of how tokens flow from the Common Pool of Gardens.**

Creation of a new [subgraph](https://thegraph.com/docs/en/developer/define-subgraph-hosted/) that keeps track of funding proposals, amounts, and beneficiaries.

Allow the creation of teams (swarms) linking them to existing proposals data.

Detection of popular DAO frameworks to further analyze individual contributors. In case the funding proposal sent funds to an Aragon Vault, Gnosis Safe, or Coloni Vault. The site will show further details about the list of contributors, addresses, amounts, and access to the user profile.

New and interesting ways to visualize that data within different timeframes.

Teams might attach updates with the deliverables they worked on in a specific period. To get more correlations on the expenditures and the outcomes in each period.

### Exploration

Community members would like to have a high-level overview of how a Garden disburses the funds on the Common Pool. And how effective a beneficiary is with the funding proposal they received. The data displayed should optimize to get more informed decisions of everyone.

### Pains

- Lack of a high-level visualization of Gardens expenditure in a timeframe.
- Difficult to assess a working group KPIs vs burning rates amounts.
- Not having enough context about the success of funding proposals.

### Test pilot

@hernandoagf did an initial work on this direction creating a [Dune dashboard for 1Hive](https://dune.xyz/hernandoagf/1Hive-DI).

@boring877 worked on an excellent report of the [Outflows and Inflows from 1Hive Common Pool](https://forum.1hive.org/t/outflows-inflows-common-pool/1754).

Both have a lot of great feedback from the community and prove to be a good signal of the usefulness of this kind of data.

## Tech spec

Doing further work on Dune to get more details about proposal expenditures showed @hernandoagf that we might hit the limit capabilities of what we can do with Dune as a generalization for every Garden. So we most likely use subgraphs to be a more powerful and extensible alternative.

Use [nivo](https://nivo.rocks/) for visualizing the data as we already have experience with including charts on the Gardens onboarding wizard.

To generalize Gardens data we can rely on the [Data Source Template](https://thegraph.com/docs/en/developer/create-subgraph-hosted/#data-source-templates) concept of subgraphs. And evaluate each of the popular DAO frameworks to analyze how funds flow from Common Pool to the contributor. We can take the [current gardens subgraph](https://github.com/1Hive/gardens/tree/master/packages/subgraph) as inspiration.

To unit test the subgraphs we could leverage the [matchstick library](https://github.com/LimeChain/matchstick).

**Comment by willjgriff** [comment]
> Use nivo for visualizing the data as we already have experience with including charts on the Gardens onboarding wizard.

Would integrating into nivo take much work or is it fairly straight forward?

This sounds like something worth having although I'm not sure if it should take priority for now, we should evaluate everything that needs doing and figure that out.

**Comment by Jeff-Emmett** [comment]
Hey guys, I have been doing some thinking as well about how to visualize and communicate relevant flows in the 1Hive & Gardens ecosystems.  (Love the Dune Analytics dashboard, by the way!)

I drew up this sketch of a possible sankey diagram (could be done with a Nivo visualization), that might help community members understand the inflow and outflow of funds in 1Hive on a monthly basis, as well as a rundown of the productive outputs produced by that community investment.

Some visualization of HNY issuance being dynamically minted into the Common Pool would be pretty cool too. 1Hive is really neat & unique in that its issuance really makes it a "production metabolism" - would be cool to see that HNY flow depicted somehow for people to more easily understand.

There are some half baked thoughts at the bottom of the diagram that may or may not be applicable.

![Sketches - page 19](https://user-images.githubusercontent.com/46964190/151893788-6472905b-6a8f-47ad-aed1-804a28f167c0.png)

If this is helpful at all, happy to chat further if so!
